### PR TITLE
[CBRD-22123] Broker failed to connect to database server

### PR DESCRIPTION
**Pull request by paulgheorghecristian**
CAS connect timed out because the whole connection process was slowed down because of thread spawning, which is relatively slow in an already thread loaded system. The solution was to start the connection threads at initialization; this leads to improved connection times.

### CODE REVIEW

File: src/base/mprec.h [hunk 1]
```diff
@@ -140,7 +140,7 @@ extern "C"
 #define Exp_msk1    ((uint32_t)0x00800000L)
 #define Exp_msk11   ((uint32_t)0x00800000L)
 #define Exp_mask    ((uint32_t)0x7f800000L)
-#define PREC        24
+#define P    	    24
```
Comment (eseokoh) [inline src/base/mprec.h]: It was Razvan's fix. Please refer to #1123

File: src/thread/thread_worker_pool.hpp [hunk 2]
```diff
@@ -1247,12 +1281,27 @@ namespace cubthread
...
-	do
+	if (m_task_p == NULL)
+	  {
+	    // started without task; get one
+	    if (get_new_task ())
+	      {
+		assert (m_task_p != NULL);
+	      }
+	  }
+	if (m_task_p != NULL)\
```
Comment (eseokoh) [inline src/thread/thread_worker_pool.hpp]: please find typo, `\` :)

File: src/thread/thread_manager.cpp [hunk 3]
```diff
@@ -140,7 +140,7 @@ namespace cubthread
 
   entry_workpool *
   manager::create_worker_pool (size_t pool_size, size_t task_max_count, entry_manager *context_manager,
-			       std::size_t core_count, bool debug_logging)
+			       std::size_t core_count, bool debug_logging, bool pool_threads, cubperf::duration transition_period)
```
Comment (ribram) [inline src/thread/thread_manager.cpp]: Rename as wait_for_task_time or fade_out_time or something more descriptive than "transition".

cubperf::duration should be reserved to performance statistics. I think std::chrono::seconds is more appropriate.

File: src/thread/thread_worker_pool.hpp [hunk 4]
```diff
@@ -1013,26 +1047,26 @@ namespace cubthread
   void
   worker_pool<Context>::core::worker::push_task_on_new_thread (task<Context> *work_p, cubperf::time_point push_time)
```
Comment (ribram) [inline src/thread/thread_worker_pool.hpp]: Rename push_task_on_new_thread as start_[new_]thread.

File: src/thread/thread_worker_pool.hpp [hunk 5]
```diff
@@ -1013,26 +1047,26 @@ namespace cubthread
...
-    m_task_p = work_p;
-
-    // start thread.
-    //
-    // the next code tries to help visualizing any system errors that can occur during create or detach in debug mode
-    //
-    // release will basically be reduced to:
-    // std::thread (&worker::run, this).detach ();
-    //
+    if (work_p != NULL)
```
Comment (ribram) [inline src/thread/thread_worker_pool.hpp]: Remove if condition and `assert (m_task_p == NULL);` safe-guard. Other operations and comments are not restricted to `m_task_p != NULL` case.

**Pull request merged**